src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.ComposeCache.targets

```diff
@@ -0,0 +1,370 @@
+    ============================================================
+    -->
+  <Target Name="PrepareForComposeCache">
+
+    <PropertyGroup>
+      <PreserveCacheLayout Condition="'$(PreserveCacheLayout)' == ''">true</PreserveCacheLayout>
+      <PublishDir>$(ComposeDir)</PublishDir>
+      <PublishDir Condition="'$(ComposeDir)' == ''">$(OutputPath)</PublishDir>
+      <PublishDir Condition="'$(ComposeDir)' == '' and '$(RuntimeIdentifier)' != ''">$(OutputPath)$(RuntimeIdentifier)\</PublishDir>
+      
+      <FX_Version Condition="'$(FX_Version)' == ''">$(_ShortFrameworkVersion)</FX_Version>
+      <_CrossProjFileDir>$([System.IO.Path]::Combine($([System.IO.Path]::GetTempPath()),"Optimize_Crossgen_$(RuntimeIdentifier)_$(TargetFramework)_$(FX_Version)"))</_CrossProjFileDir>
+      <!-- Ensure any PublishDir has a trailing slash, so it can be concatenated -->
+      <PublishDir Condition="!HasTrailingSlash('$(PublishDir)')">$(PublishDir)\</PublishDir>
+    </PropertyGroup>
+
+    <MakeDir Directories="$(PublishDir)" />
+
+  </Target>
+
+  <!--
+    ============================================================
+                                        _CleanupTempFiles
+
+    Cleans up the temp directory used by optimize phase
+    ============================================================
+    -->
+  <Target Name="_CleanupTempFiles"
+          Condition="'$(PreserveCrossGenTempDir)'==''" >
+    <RemoveDir  
+        Directories="$(_CrossProjFileDir)" />  
+  </Target>
+  <!--
+    ============================================================
+                                        ComputeAndCopyFilesToCacheDirectory
+
+    Computes the list of all files to copy to the publish directory and then publishes them.
+    ============================================================
+    -->
+  <Target Name="ComputeAndCopyFilesToCacheDirectory"
+          DependsOnTargets="ComputeFilesToCache;
+                            CopyFilesToCacheDirectory" />
+
+  <!--
+    ============================================================
+                                        CopyFilesToCacheDirectory
+
+    Copy all build outputs, satellites and other necessary files to the publish directory.
+    ============================================================
+    -->
+  <Target Name="CopyFilesToCacheDirectory"
+          DependsOnTargets="_CopyResolvedManagedFiles;
+                            _CopyResolvedUnManagedFiles;
+                            _CleanupTempFiles">
+
+    <Message Text="Files were composed in $(PublishDir)"
+             Importance="high"/>
+  </Target>
+
+  <!--
+    ============================================================
+                                        _CopyResolvedManagedFiles
+
+    Copy _ManagedResolvedFileToPublish items to the publish directory.
+    ============================================================
+    -->
+  <Target Name="_CopyResolvedManagedFiles"
+          DependsOnTargets="_ComputeResolvedFilesToCachTypes;
```

Comment:
It looks like there's an 'e' missing in "Cache" for `_ComputeResolvedFilesToCachTypes`